chat/attachment.py

```diff
@@ -0,0 +1,275 @@
+from __future__ import annotations
+
+import math
+import base64
+
+from hikari import Attachment
+from abc import ABCMeta, abstractmethod
+from typing import List
+from PIL import Image
+from io import BytesIO
+
+from utils import humanize_tuple
+from options import ATTACHMENT_POLICIES as GLOBAL_OPTIONS
+
+
+class AttachmentGroup:
+    def __init__(
+        self, *, texts: list[TextAttachment], images: list[GPTImageAttachment]
+    ):
+        self.texts = texts
+        self.images = images
+
+    async def export(self, text_content: str) -> List[dict]:
+        content = []
+
+        text = text_content
+        if len(self.texts) == 0:
+            pass
+        elif len(self.texts) == 1:
+            text += "\n\n" + await self.texts[0].digest()
+        else:
+            for txt in self.texts:
+                text += f"\n\n{txt.attachment.filename}:\n" + await txt.digest()
+        if text:
+            content.append({"type": "text", "text": text})
+
+        for img in self.images:
+            content.append(await img.digest())
+
+        return content
+
+
+class DiscordAttachment(metaclass=ABCMeta):
+    def __init__(self, attachment: Attachment):
+        self._attachment = attachment
+
+    @property
+    def attachment(self):
+        return self._attachment
+
+    @property
+    def size(self):
+        return self.attachment.size
+
+    @abstractmethod
+    def check_error(self) -> Exception | None:
+        pass
+
+    @abstractmethod
+    async def digest(self):
+        pass
+
+
+class TextAttachment(DiscordAttachment):
+    MAX_SIZE, ALLOWED_EXTENSIONS = map(
+        GLOBAL_OPTIONS.get, ("max_text_file_size", "allowed_text_extensions")
+    )
+
+    def check_error(self):
+        att = self.attachment
+        if not att.filename.endswith(self.ALLOWED_EXTENSIONS):
+            return ValueError(
+                f"Only supported extensions in text files are {humanize_tuple(self.ALLOWED_EXTENSIONS)}."
+            )
+        elif att.size > self.MAX_SIZE:
+            return ValueError(
+                f"File is too large to upload. It must be at most {self.MAX_SIZE // 1024} KB."
+            )
+
+        return None
```

Comment:
`GLOBAL_OPTIONS.get`를 객체 클래스에 저장하는 대신 settings 클래스를 추가로 선언해서 사용하는게 좋겠습니다.

예시

```suggestion
class Options(BaseModel):
    max_text_file_size: int = 4096  # 값은 예시
    allowed_text_extensions: bool = False


options = Options()


class TextAttachment(DiscordAttachment):
    def check_error(self):
        att = self.attachment
        if not att.filename.endswith(settings.allowed_text_extensions):
            return ValueError(
                f"Only supported extensions in text files are {humanize_tuple(self.ALLOWED_EXTENSIONS)}."
            )
        elif att.size > settings.max_text_file_size:
            return ValueError(
                f"File is too large to upload. It must be at most {self.MAX_SIZE // 1024} KB."
            )

        return None
```